openedx/tests/xblock_integration/xblock_testcase.py

```diff
@@ -379,6 +413,54 @@ def _get_handler_url(self, handler, xblock_name=None):
             'suffix': ''
         })
 
+    def extract_block(self, content, urlname):
+        '''This will extract the HTML of a rendered XBlock from a
+        page. This should be simple. This should just be (in lxml):
+            usage_id = self.xblocks[block_urlname].scope_ids.usage_id
+            encoded_id = usage_id.replace(";_", "/")
+        Followed by:
+            page_xml = defusedxml.ElementTree.parse(StringIO.StringIO(response_content))
+            page_xml.find("//[@data-usage-id={usage}]".format(usage=encoded_id))
+        or
+            soup_html = BeautifulSoup(response_content, 'html.parser')
+            soup_html.find(**{"data-usage-id": encoded_id})
+
+        Why isn't it? Well, the blocks are stored in a rather funky
+        way in learning sequences. Ugh. Easy enough, populate the
+        course with just verticals. Well, that doesn't work
+        either. The whole test infrastructure populates courses with
+        Studio AJAX calls, and Studio has broken support for anything
+        other than course/sequence/vertical/block.
+
+        So until we either fix Studio to support most course
+        structures, fix learning sequences to not have HTML-in-JS
+        (which causes many other problems as well -- including
+        user-facing bugs), or fix the test infrastructure to
+        create courses from OLX, we're stuck with this little hack.
+        '''
+        usage_id = self.xblocks[urlname].scope_ids.usage_id
+        # First, we get out our <div>
+        soup_html = BeautifulSoup(content)
+        xblock_html = unicode(soup_html.find(id="seq_contents_0"))
```

Comment:
So you are assuming the xblock is rendered in the first sequential (and is the only thing there)? If so, the expected preconditions should be called out in the documentation.

Should you at least look to see if the usage_id is present in that element and error if not?